### DISCUSSION TITLE
Port to use geometry nodes?

### DISCUSSION THREAD

**Discussion by paulmelis** [tkeskita/BVtkNodes#75] [Ideas]
Hi, this node set currently appears to use a custom editor page. With the new Geometry Nodes in Blender becoming the go-to way to generate geometry in a flexible way, do you plan to see if vtk nodes can be ported to GN?

**Comment by tkeskita** [comment]
Hi Paul!

I haven't looked at Geometry Nodes yet, so I don't know it's possibilities. What would be the benefits of GN, in your view? 

Do you think GN is stable enough to be tied with VTK? Lately upgrades to new Blender and VTK versions have been getting more painful, so I'm slightly worried about the upkeep costs..

**Reply by paulmelis** [reply]
> What would be the benefits of GN, in your view?

Well, the VTK nodes could potentially have a better fit in Blender with GN. Right now they're pretty much separated from the rest, while most of the VTK nodes operate on geometry (or data that gets turned into geometry at some point) and so could possibly fit into the GN framework well. 

> Do you think GN is stable enough to be tied with VTK? Lately upgrades to new Blender and VTK versions have been getting more painful, so I'm slightly worried about the upkeep costs..

There still is quite a bit of development going on around GN, that's true. With the upcoming 3.1 a few more GN essentials will be fleshed out more, but probably some other developments will happen after that. What exactly are the challenges in upgrading VTK+Blender lately? Is the coupling done purely in Python, or also in C/C++?

**Reply by tkeskita** [reply]
Thanks for comments! I'll try to check GN at some point, that seems interesting. Also, if anyone is interested in this or maybe even tried something, please feel free to comment. Identification of benefits for going for GN are most welcome!

> What exactly are the challenges in upgrading VTK+Blender lately? Is the coupling done purely in Python, or also in C/C++?

The coupling is pure Python. Most headache originated from Python upgrade from 3.7 to 3.9, but there were also a few changes on Blender Python API and VTK which needed to be aligned. There might still be some bugs introduced in the upgrade to customized nodes, which may be revealed down the line..

**Reply by renehommel** [reply]
Even though I am not a developer, but I am very familiar with BVTK and allow me to comment on this topic:

I do not agree. The possible integration builds another barrier to entry. Besides, GN are still quite new and it is difficult to see where their journey will take them. Most important for us, however, is the point of purpose: GN have the task of enabling parametric design. BVTK has the task to provide high level visualization. Why should this concept change? What was the perfectly working Unix philosophy?  "Do One Thing and Do It Well."

And this is exactly where BVTK is best positioned. Thanks to the great effort of its developers, which I and many others I know well appreciate.
